### DISCUSSION TITLE
MetaMask disconnect can be supported now

### DISCUSSION THREAD

**Discussion by davisshaver** [wevm/wagmi#3909] [Ideas]
### Describe the bug

Previously we had https://github.com/wevm/wagmi/issues/684 which was closed because MetaMask did not support [programmatic disconnect](https://github.com/MetaMask/metamask-extension/issues/10353#issuecomment-1011129684). MetaMask recently added a `wallet_revokePermissions` RPC method which allows revoking permissions.

See also tracking issue here https://github.com/MetaMask/core/issues/723.

 With this change, could WAGMI be updated to "fix" the issue with MetaMask not fully disconnecting when use WAGMI disconnect hooks?

### Wagmi Version

^2.8.1

### Viem Version

^2.9.31

### Check existing issues

- [X] I checked there isn't [already an issue](https://github.com/wevm/wagmi/issues) for the bug I encountered.

### Anything else?

Thank you for a fantastic library and please advise if I can help with this issue followup in any way.

**Comment by tmm** [comment]
Moved to discussion as this is a feature request

**Comment by tmm** [comment]
Opened https://github.com/wevm/wagmi/pull/3910

**Comment by tmm** [comment]
Released!
